x-pack/plugins/fleet/common/authz.ts

```diff
@@ -109,7 +117,12 @@ export const calculateAuthz = ({
         allSettings: fleet.settings?.all ?? false,
         allAgentPolicies: fleet.agentPolicies?.all ?? false,
         addAgents: fleet.agents?.all ?? false,
-        addFleetServers: (fleet.agents?.all && fleet.settings?.all) ?? false,
+        addFleetServers:
+          (fleet.agents?.all &&
+            fleet.settings?.all &&
+            fleet.agentPolicies?.all &&
```

Comment:
@jen-huang @nimarezainia This simplify a lot the permissions check, if we only in allow in the UI to: "add fleet server" if they can also add agent policies, as we need to create an agent policy fleet server, do we think it's something that make sense? To rephrase it does it make sense to only show add fleet server for "agents:all, settings:all, agentPolicies:all"

Reply:
Edit: actually I did the same thing as on the add agent flyout if a user cannot create a policy and just disabled the form
<img width="823" alt="Screenshot 2024-04-29 at 3 33 34 PM" src="https://github.com/elastic/kibana/assets/1336873/6c2b15ce-2124-4ead-bd83-0406578e8aa0">